File: my_robot.rb
class MyRobot < Alicorn::Sphere
    def create
        @sdk = getSphere("AndroidFTC")
        @joy1 = @sdk.get("gamepad_1")
    end

    def update
        x = @joy1.get("x2").invert()
        y = @joy1.get("y1")

        left = x.subtract(y).require(0.2)
        right = x.add(y).require(0.2)

        #@sdk.set("left", left)
        #@sdk.set("right", right)

        @sdk.set("left", @joy1.get("y1"))
        @sdk.set("right", @joy1.get("y2"))
    end
end

File: MyRobot.js
var sdk, joy1;
alicorn.shard("MyRobot")
    .create(function(self) {
        sdk = self.require("AndroidFTC");
        joy1 = sdk.get("gamepad_1");
    })
    .update(function(self) {
        var y2 = joy1.get("y2");
        var y = joy1.get("y1");

        var left = y2.invert() / 2;
        var right = y / 2;

        sdk.set("left", left);
        sdk.set("right", right);
    });
